### PR TITLE
Scf guess test

### LINKED ISSUE #3040: Add more strenuous tests for SCF initial guesses
Currently the scf-guess suite only checks final energies, which is simple but also not very useful. There should be a variant of the tests that check the energy of the first iteration, which will tell us if a test gets broken or not.

### PR DESCRIPTION
**Pull request by Awallace3**
## Description

closes #3040

## User API & Changelog headlines

- [ ] RN 1
- [ ] RN 2

## Dev notes & details

- [x] print_variables prints array variables. (first and last element)
- [x] Used this to test scf guesses in pytest.
- [x] Store new qc variable "SCF TOTAL ENERGIES" that is a list of all the SCF iteration energies.

## Questions
- [ ] Question1

## Checklist
- [x] Tests added for any new features
- [ ] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [ ] Ready for merge

### CODE REVIEW

File: psi4/driver/procrouting/scf_proc/scf_iterator.py [hunk 1]
```diff
@@ -80,6 +80,7 @@ def scf_compute_energy(self):
         self.initialize_jk(self.memory_jk_)
     else:
         self.initialize()
+    self.iterene = []
```
Comment (susilehtola) [inline psi4/driver/procrouting/scf_proc/scf_iterator.py:83]: Use a more descriptive variable name, e.g.
```suggestion
    self.iteration_energies = []
```

File: psi4/driver/procrouting/scf_proc/scf_iterator.py [hunk 2]
```diff
@@ -680,6 +683,12 @@ def scf_finalize_energy(self):
     core.print_out("\n\n")
     self.print_energies()
 
+    # reshape to be 2-d
```
Comment (susilehtola) [inline psi4/driver/procrouting/scf_proc/scf_iterator.py:686]: Confusing comment; I think you mean that the vector of length `N` is returned as a `N x 1` matrix?

File: tests/pytests/test_scf_options.py [hunk 3]
```diff
@@ -38,7 +43,99 @@ def test_guess_mix_for_broken_symmetry(inp):
...
+            "reference": ref,
+            "basis": "cc-pvtz",
+            "scf_type": "pk",
+            "df_scf_guess": False,
+            "guess": inp["options"]["guess"],
+        }
+    if ref == "rhf":
+        hf = psi4.geometry(
+            """
+    0 1
```
Comment (davpoolechem) [inline tests/pytests/test_scf_options.py:114]: I know the spacing here was your autoformatter's fault, but probably better to have this indented properly.

File: tests/pytests/test_scf_options.py [hunk 4]
```diff
@@ -38,7 +43,99 @@ def test_guess_mix_for_broken_symmetry(inp):
...
+    F
+    H 1 0.9015
+    """
+        )
+        options["docc"] = [3, 0, 1, 1]
+    else:
+        hf = psi4.geometry(
+            """
+    1 2
```
Comment (davpoolechem) [inline tests/pytests/test_scf_options.py:123]: Also here, with regards to indenting/spacing.

File: tests/pytests/test_scf_options.py (hunk 3 above)
Comment (loriab) [reply tests/pytests/test_scf_options.py:114]: Mol formatting is pretty resilient.

**Review by loriab — APPROVED** [review]
lgtm, thanks!

**Review by jturney — APPROVED** [review]

**Pull request merged**